src/common/io.py

```diff
@@ -1382,37 +1388,48 @@ def _read_trajectory_data(self, data_index, read_diag_data):
         nbr_diag_points    = int(self._data_3_info.shape[0])
         nbr_diag_variables = int(self._data_4_info.shape[0])
 
+        # TODO account for start_index and stop_index
 
         data, self._file_pos_model_var, self._file_pos_diag_var = fmi_util.read_diagnostics_trajectory(
-                                                encode(self._fname), int(read_diag_data), int(self._has_file_pos_data),
-                                                self._file_pos_model_var, self._file_pos_diag_var,
-                                                data_index, file_position, sizeof_type, nbr_points, nbr_diag_points,
-                                                nbr_variables, nbr_diag_variables)
+            encode(self._fname),
+            int(read_diag_data),
+            int(self._has_file_pos_data),
+            self._file_pos_model_var,
+            self._file_pos_diag_var,
+            data_index,
+            file_position,
+            sizeof_type,
+            nbr_points,
+            nbr_diag_points,
+            nbr_variables,
+            nbr_diag_variables
+        )
         self._has_file_pos_data = True
 
         return data
 
-    def _get_interpolated_trajectory(self, data_index):
+    def _get_interpolated_trajectory(self, data_index: int, start_index: int = None, stop_index: int = None) -> Trajectory:
         """ Returns an interpolated trajectory for variable of corresponding index 'data_index'. """
         self._verify_file_data()
-        
+
         if data_index in self._data_2:
             return self._data_2[data_index]
-        
-        diag_time_vector = self._get_diagnostics_trajectory(0)
-        time_vector      = self._read_trajectory_data(0, False)
-        data             = self._read_trajectory_data(data_index, False)
-        
+
+        diag_time_vector = self._get_diagnostics_trajectory(0, start_index, stop_index)
+        time_vector      = self._read_trajectory_data(0, False, start_index, stop_index)
+        data             = self._read_trajectory_data(data_index, False, start_index, stop_index)
+
         f = scipy.interpolate.interp1d(time_vector, data, fill_value="extrapolate")
-        
-        self._data_2[data_index] = f(diag_time_vector)
-        
-        return self._data_2[data_index]
 
+        self._data_2[data_index] = f(diag_time_vector)
+        return self._data_2[data_index][start_index:stop_index]
```

Comment:
Maybe add a comment what happens here with the slicing for the default values of start_index, stop_index as None? This is not obvious

Reply:
I thought about it but since we have this in 7 places right now with `[start_index:stop_index]` I stopped myself from writing that comment. Let me know what you think. I also agree it wasn't obvious to me either but it makes perfect sense after testing it.

Reply:
I think it still won't be obvious to the next person stumbling upon this, including us in > 2 months; I'd suggest to just add it, even it requires to be in a few places.

Reply:
What about in the docstring for `get_variables_data` where the arguments `start` and `stop` are exposed, we explain that `start` and `stop` acts like slices, i.e. `[start:stop]`, and clarify that inputs such as `[None:None]` or `[5:None]` are all valid calls? And we could even refer to https://docs.python.org/3/library/functions.html#slice where it says that `start` and `stop` default to `None`?

Reply:
I think in the docstrings + that start, stop act as slices with `None` being no restriction sounds good.

Reply:
It was already part of the docstring but I made it even more clear now